src/components/icon/icon.js

```diff
@@ -55,6 +60,119 @@ const IconType = {
  * @extends {Component}
  */
 class Icon extends Component {
+  _defaultColor: string = style.white;
+  _activeColor: string = style.brandColor;
+  _className: string = '';
+
+  /**
+   * @constructor
+   * @param {Object} props - component props
+   */
+  constructor(props: Object) {
+    super(props);
+    const {path, id} = props;
+    if (path && id) {
+      this._className = `playkit-icon-${id}`;
+      // Avoid from override existing classes
+      if (!document.getElementsByClassName(this._className).length) {
+        this.createDynamicIconClass(props);
+      }
+    }
+  }
+
+  /**
+   * Creates a dynamic icon class with background image settings
+   *
+   * @param {Object} props - component props
+   * @returns {void}
+   * @memberof Icon
+   */
+  createDynamicIconClass = (props: Object) => {
+    const {path, state, color, activeColor} = props;
+    const fillColor = this.getFillColor(state, color, activeColor);
+    const pathTag = this.getPathTag(path, fillColor);
+    const svgUrl = this.getSVGUrl(pathTag);
+    const css = `.${this._className} { background-image: ${svgUrl}; }`;
+    const style = document.createElement('style');
+    style.type = 'text/css';
+    document.getElementsByTagName('head')[0].appendChild(style);
+    style.appendChild(document.createTextNode(css));
```

Comment:
maybe use one stylesheet and add rules to it to avoid cluttering the head section
https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule